lib/agenda/every.js

```diff
@@ -3,80 +3,61 @@ const debug = require('debug')('agenda:every');
 
 /**
  * Creates a scheduled job with given interval and name/names of the job to run
+ * @name Agenda#every
+ * @function
  * @param {Number} interval run every X interval
  * @param {*} names String or strings of jobs to schedule
  * @param {Object} data data to run for job
  * @param {Object} options options to run job for
- * @param {Function} cb called when schedule fails or passes
- * @returns {*} Job or jobs created
+ * @returns {Promise} Job/s created. Resolves when schedule fails or passes
  */
-module.exports = function(interval, names, data, options, cb) {
+module.exports = async function(interval, names, data, options) {
```

Comment:
Worth noting that this means the new agenda will require node >= 7.6.0. I don't think that's the right choice since Node 6 [won't be EOL-ed until April 2019](https://github.com/nodejs/Release) so it's a little early to drop support for it.

Reply:
We could always add an async pollyfill such as https://github.com/touskar/async-await-es7?

Reply:
Also keep in mind this will be pushed to version `3.0.0` so people can still use version `2.0.0` if they don't want promise support.

Reply:
I'd suggest we just use [co](https://www.npmjs.com/package/co), works with node v4 with no transpilers and all you need to do is replace `async function() { /* ... */ }` with `co.wrap(function*() { /* ... */ });` and `await` with `yield`.

Reply:
Since it's using yield doesn't that mean instead of using native async we'd be using generators? I thought they were a lot slower?

Reply:
I'm personally against using a wrapper library when we could just drop support for older node versions. Anything below 7 or more specifically 8 can just use the v2 branch with callbacks, etc.

I'm not being on another project where we just sit back and don't progress because a few places that use this work for *free* can't/won't upgrade.

Reply:
I haven't seen anything about them being noticeably slower, been using both co and async/await in prod for years, including with agenda 0.9. If you're ok with me making the changes to switch from async/await to co internally, I'll do it. I'd be all for dropping support for node 4 and 6 if it added significant developer overhead, except for in this case the effort necessary to support node 4 and 6 is so marginal it seems silly to not support it.

Reply:
@agenda/maintainers thoughts?

Reply:
If you're willing to look after any async(co) wrapper code for the foreseeable future I'm okay with it.We'll need to make sure to update the jsdoc then. Any ideas on how co is represented in a jsdoc?

Reply:
`const fn = co.wrap(function*(){});` -> `fn()` returns a promise

Reply:
@vkarpov15 join us on Slack so we can discuss.